app/ota_client.py

```diff
@@ -105,50 +172,55 @@ def _path_to_url(base: str, p: Union[Path, str]) -> str:
             pass
 
         quoted_path = quote_from_bytes(bytes(relative_path))
-        return urljoin(base, quoted_path)
 
+        # switch scheme if needed
+        _url_parsed = urlparse(urljoin(base, quoted_path))
+        # unconditionally set scheme to HTTP if proxy is applied
+        if self._proxy_set:
+            _url_parsed = _url_parsed._replace(scheme="http")
+
+        return _url_parsed.geturl()
+
+    @partial(_retry, RETRY_COUNT, OUTER_BACKOFF_FACTOR, BACKOFF_MAX)
     def __call__(
-        self, url_base: str, path: str, dst: Path, digest: str, cookies: Dict[str, str]
+        self,
+        url_base: str,
+        path: str,
+        dst: Path,
+        digest: str,
+        cookies: Dict[str, str],
+        headers: Dict[str, str] = None,
     ) -> int:
         url = self._path_to_url(url_base, path)
+        if not headers:
+            headers = dict()
 
-        error_count = 0
         try:
-            response = self._session.get(url, stream=True, timeout=10, cookies=cookies)
+            error_count = 0
+            response = self._session.get(
```

Comment:
Let me confirm, this session.get is retried by RETRY_COUNT times with retry_strategy and this __call__ function is also retried with `@partial(_retry, RETRY_COUNT, OUTER_BACKOFF_FACTOR, BACKOFF_MAX)` ?
I think it is ok since more robust.

Reply:
Yes! The outer retry will also retry on the failure of inner `urllib3.Retry`, it is a little bit redundant but I think it is not a very big problem.